providers/implementations/serializers/serializer_common.c

```diff
@@ -172,44 +172,53 @@ int ossl_prov_print_labeled_bignum(BIO *out, const char *label,
     if (ossl_prov_bio_printf(out, "%s%s\n", label, neg) <= 0)
         return 0;
 
-    /* Skip past the zero bytes in the first word */
-    for (off = 0; off < BN_BYTES; off++) {
-        BN_ULONG l = words[0];
-        int o = 8 * (BN_BYTES - off - 1);
-        int b = ((l & (0xffLU << o)) >> o) & 0xff;
-
-        if (b != 0)
-            break;
-    }
-    /* print 16 hex digits per line, indented with 4 spaces */
-    for (i = 0; i < bytes; i += 16) {
-        int j, step;
-
-        if (ossl_prov_bio_printf(out, "    ") <= 0)
-            return 0;
-
-        for (j = 0; i + j < bytes && j < 16; j += BN_BYTES - step) {
-            int k;
-            BN_ULONG l = words[(i + j + off) / BN_BYTES];
-
-            step = (i + j + off) % BN_BYTES;
-
-            for (k = step; k < BN_BYTES; k++) {
-                int o = 8 * (BN_BYTES - k - 1);
-                int b = ((l & (0xffLU << o)) >> o) & 0xff;
+    /* Keep track of how many bytes we have printed out so far */
```

Comment:
Could this just use BN_bn2hex instead and then print it out formatted..? (It does an alloc but it would be much simpler)

Reply:
BN_bn2hex creates one long line.  We *still* need to keep track of how many bytes are printed out so far, when to insert the colon and all that.  The difference will be very little, at the cost of an unnecessary malloc().  I'm not sure there's much to gain with that...